app/src/main/java/com/grarak/kerneladiutor/fragments/other/SettingsFragment.java

```diff
@@ -324,7 +324,7 @@ private void dumpsysfs() {
         // loop through each array in the constants file. These contain all the other arrays.
         // have to do this once for the 1d arrays and again for the 2 arrays
         try {
-            File sysfsdump = new File(MainActivity.context.getFilesDir(), "sysfsdump.txt");
+            File sysfsdump = File.createTempFile("sysfsdump", ".txt", MainActivity.context.getExternalMediaDirs()[0]);
```

Comment:
Yah, its kinda tricky. You need to use a file provider to share the file with other apps.